pkg/operator/controller.go

```diff
@@ -344,8 +412,9 @@ func (c *Controller) handler() cache.ResourceEventHandlerFuncs {
 	}
 }
 
-func (c *Controller) Run(stopCh <-chan struct{}) error {
+func (c *Controller) Run(stopCh chan struct{}) error {
```

Comment:
Why remove `<-`? I thought you could define a member like so:

```
type MyChannels struct {
  stopCh <-chan struct{}
}
```

Reply:
This was my first attempt on ending the Operator(I see that if i close that channel it stops). The fact is that i can't close a `<-chan`, only if it is `chan`, go did not allow me to close a read only channel.

Reply:
I will resolve this one, my initial approach was just a sketch and the final solution will not involve keeping track of `stopCh`.

Reply:
I believe that you can still use close(chan) to close the channe (notice not using the <-)l, the <-just signifies that it is a read only channel

Reply:
Well, the problem is that i was willing to keep a reference to the channel within the struct so i could close it later on. If you look at this sample below you gonna see what i mean, i needed to keep the reference as `chan` instead of `<-chan`.

https://goplay.space/#qj9_WWu4izg

Reply:
Not that a reference to a `<-chan` is not valid but if i use it i can't use `close()`. Go is so clever :-D